src/main/java/com/ericsson/eiffel/remrem/generate/cli/CLI.java

```diff
@@ -0,0 +1,137 @@
+package com.ericsson.eiffel.remrem.generate.cli;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import com.ericsson.eiffel.remrem.semantics.SemanticsService;
+import com.ericsson.eiffel.remrem.shared.MsgService;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * @author evasiba
+ *
+ */
+public class CLI {
+    private Options options=null;
+
+    public CLI() {
+        options = createCLIOptions();
+    }
+
+    /**
+     * Creates the options needed by command line interface
+     * @return the options this CLI can handle
+     */
+    private static Options createCLIOptions() {
+        Options options = new Options();
+        options.addOption("h", "help", false, "show help.");
+        options.addOption("f", "content_file", true, "message content file");
+        options.addOption("json", "json_content", true, "json content");
+        options.addOption("t", "message_type", true, "message type, mandatory if -f or -json is given");
+        options.addOption("r", "response_file", true, "file to store the response in, optional");
+        return options;
+    }
+
+    /**
+     * Prints the help for this application and exits.
+     * @param options the options to print usage help for
+     */
+    private static void help(Options options) {
+        // This prints out some help
+        HelpFormatter formater = new HelpFormatter();
```

Comment:
I think we should add a header specifying that without options it will start as a service. If any options are provided it will just process those without starting a service.